src/nvim/terminal.c

```diff
@@ -250,12 +253,16 @@ Terminal *terminal_open(TerminalOptions opts)
   rv->sb_size = MIN(rv->sb_size, 100000);
   rv->sb_buffer = xmalloc(sizeof(ScrollbackLine *) * rv->sb_size);
 
+  if (!true_color) {
+      return rv;
+  }
+
   // Configure the color palette. Try to get the color from:
   //
   // - b:terminal_color_{NUM}
   // - g:terminal_color_{NUM}
   // - the VTerm instance
```

Comment:
> Is "Only set custom terminal colors if in true color mode" enough?

Don't bother; the if-statement communicates that clearly enough.

I'm not sure about color_indexes, I'll take your word for it. Does this have any bearing on the following:

https://github.com/neovim/neovim/issues/3459

https://github.com/neovim/neovim/issues/2323